gateway/engine/es/src/main/java/io/apiman/gateway/engine/es/ESMetrics.java

```diff
@@ -96,13 +110,22 @@ public void run() {
      */
     protected void processQueue() {
         try {
-            RequestMetric metric = queue.take();
-            Index index = new Index.Builder(metric).refresh(false)
-                    .index(getIndexName())
-                    .type("request").build(); //$NON-NLS-1$
-            DocumentResult result = getClient().execute(index);
+            Collection<RequestMetric> batch = new ArrayList<>(this.batchSize);
+            RequestMetric rm = queue.take();
+            batch.add(rm);
+			queue.drainTo(batch, this.batchSize - 1);
+			
+			Builder builder = new Bulk.Builder();
```

Comment:
Gotcha, I see. It always runs as quickly as it can, and sends a batch of up-to a given size (rather than waiting for the size to accumulate to a specific target and/or time).

Sounds good, thanks!